### PR TITLE
Empty check name template was causing an error

### LINKED ISSUE #43: Whenever we run this check, it throws error: creating event
Whenever we execute the sensu-check-log, it throws an error. Here's the image for reference
[error_screenshot](https://github.com/sensu/sensu-check-log/assets/153795426/793b60d7-fc4f-43c5-9007-142229342833)

Additionally, Here are the few item you would require to replicate the issue
1. Log file : [sample.log](https://github.com/sensu/sensu-check-log/files/14538884/sample.log)
2. Check_Command: `sensu-check-log --log-file /var/tmp/sample.log --match-expr "java.lang.OutOfMemoryError" --state-directory /var/tmp/ --reset-state --critical-threshold 1 --critical-only --warning-threshold 0 --ignore-initial-run --verbose`
3. Check_config: [check.yml.txt](https://github.com/sensu/sensu-check-log/files/14538931/check.yml.txt)

### PR DESCRIPTION
**Pull request by SudhanshuBawane**
closed #43 

## Description
- Fix for the issue of sensu-check-log runtime error.

## Changed
- The check name was getting passed as empty now it is fixed.

### CODE REVIEW

File: event.go [hunk 1]
```diff
@@ -13,7 +13,11 @@ func createEvent(inputEvent *corev2.Event, status int, checkNameTemplate string,
 		return nil, errors.New("negative status")
 	}
 	// Let's construct the check name from template
-	checkName, err := templates.EvalTemplate("check-name", checkNameTemplate, inputEvent)
+	var checkName = inputEvent.Check.Name
+	var err error
+	if len(checkNameTemplate) > 0 {
+		checkName, err = templates.EvalTemplate("check-name", checkNameTemplate, inputEvent)
+	}
 	if err != nil {
```
Comment (fguimond) [inline event.go:21]: I believe the error check can be moved inside the `if` block as the error will always be null if we don't call `EvalTemplate`.

File: http_test.go [hunk 2]
```diff
@@ -16,7 +16,53 @@ type testHandler struct {
...
+			expectedError:     false,
+		},
+		{
+			name:              "Invalid Template",
+			event:             sensu.FixtureEvent("entity1", "check1"),
+			checkNameTemplate: "{{ .Foo }}", // Assuming 'Foo' isn't a valid field
+			expectedCheckName: "",           // Template evaluation should fail
+			expectedError:     true,
+		},
+		// ... add more test cases with different scenarios
```
Comment (fguimond) [inline http_test.go:41]: We probably want more test here? One I can think of is with an empty ``checkNameTemplate`.
